woocommerce/woocommerce-android
# Issue/12021 handle saving product

Closes: #12021


### Description

This PR adds logic for: 
- Saving the generated product as draft
- Publishing the generated product directly

### Testing information
1. Log into an atomic or Jetpack AI enabled site
2. Navigate to the products tabs
3. Add product using AI
4. Select and image and scan the text from it
5. Click "Generate product details" 
6. Once the product is generated, select a variant of your choosing and click "Save product draft"
7. Check the product is created as "draft" and all the fields and image match with the selected ones during the flow. 
8. Repeat steps 5 to 7 but this time click on "Publish" and check the product is created and published directly. 

### Images/gif


https://github.com/user-attachments/assets/764865db-c538-4ebe-ab9b-f569cfd3e227
